tests/autoyast/validate_salt_formula.pm

```diff
@@ -0,0 +1,36 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Make sure we are the salt formula has set message-of-the-day(motd)
+# Maintainer: ybontakis <ybonatakisz@suse.com>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+
+sub run {
+    my ($self) = @_;
+    #assert_screen("login-console-motd-message") if get_var("SALT_FORMULAS_PATH");
+    assert_script_run "grep \"Happy Testing Changed\" /etc/motd";
+}
+
+sub post_fail_hook {
+    my $self = shift;
+    upload_logs('/etc/motd');
```

Comment:
let's call SUPER here to get logs from parent as well when overriding the method.

Reply:
the reason i do not use super is because so a small check as this one you collect unnecessary logs and the post_fail_hook takes a long time. if there is a problem the `console.pm` is very likely to fail first and there a complete set of logs is used

Reply:
you are assuming that console will fail, but it is really unknown why could potentially fail in the future, if we want to have any logs to attach to a bug and we want to get them in the closer location to the problem we need always to run the parent in and override of the post_fail_hook saving time to recover them manually. It is unnecessary time when you develop the test, which you can avoid with `_SKIP_POST_FAIL_HOOKS=1 but in OSD is justified that.

Reply:
why you delete completely it at the end? it was a good idea to upload it, but always you keep the `post_fail_hook` from parent.

Reply:
i had added the post_fail_hook to override the default and specialize it to the module. the full stack of the logs is more appropiate as we expect any cause of failure and such the default is just fine. In case that the cause is actually the motd texting the /etc/motd is not necessary. the test itself checks the existence and the context of it